File: templates/ios/HomeScreen.swift
import SharedTypes
import SwiftUI

struct HomeScreen: View {
    let viewModel: HomeView
    let onEvent: (Event) -> Void

    var body: some View {
        VStack(spacing: 24) {
            Spacer()
            Text(viewModel.message)
                .font(.title)
                .multilineTextAlignment(.center)
                .padding(.horizontal)
            Spacer()
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
}

#Preview {
    HomeScreen(
        viewModel: HomeView(message: "Hello from Counter"),
        onEvent: { _ in }
    )
}

File: templates/android/HomeScreen.kt
package __ANDROID_PACKAGE__.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import __ANDROID_PACKAGE__.Event
import __ANDROID_PACKAGE__.HomeView

@Composable
fun HomeScreen(
    viewModel: HomeView,
    @Suppress("UNUSED_PARAMETER") onEvent: (Event) -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(horizontal = 24.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(24.dp)
    ) {
        Spacer(modifier = Modifier.weight(1f))
        Text(
            text = viewModel.message,
            style = MaterialTheme.typography.headlineMedium,
            textAlign = TextAlign.Center,
            color = MaterialTheme.colorScheme.onBackground
        )
        Spacer(modifier = Modifier.weight(1f))
    }
}

@Preview(showBackground = true)
@Composable
fun HomeScreenPreview() {
    HomeScreen(
        viewModel = HomeView(message = "Hello from __APP_NAME__"),
        onEvent = { }
    )
}
